lib/collection/src/collection/state_management.rs

```diff
@@ -64,18 +72,66 @@ impl Collection {
     }
 
     async fn apply_config(&self, new_config: CollectionConfig) -> CollectionResult<()> {
-        log::warn!("Applying only optimizers config snapshot. Other config updates are not yet implemented.");
-        self.update_optimizer_params(new_config.optimizer_config)
-            .await?;
+        let recreate_optimizers;
 
-        // Update replication factor
         {
             let mut config = self.collection_config.write().await;
-            config.params.replication_factor = new_config.params.replication_factor;
-            config.params.write_consistency_factor = new_config.params.write_consistency_factor;
+
+            if let Err(err) = config.params.check_compatible(&new_config.params) {
+                // Stop consensus with a service error, if new config is incompatible with current one.
+                //
+                // We expect that `apply_config` is only called when configs are compatible, otherwise
+                // collection have to be *recreated*.
+                return Err(CollectionError::service_error(err.to_string()));
+            }
```

Comment:
Just to be sure: we report an error, but don't rely on this error to actually recreate the collection am I right? This is more of a sanity check so that we don't actually use this function if it wouldn't work.

Reply:
Confirmed myself: correct.